client/internal/bootstrap/grpc_test.go

```diff
@@ -8,142 +8,128 @@ import (
 	"os"
 	"path/filepath"
 	"time"
+	"testing"
+	"strings"
 
-	. "github.com/onsi/ginkgo/v2"
-	. "github.com/onsi/gomega"
 	"google.golang.org/grpc/test/bufconn"
 
 	"github.com/Azure/aks-secure-tls-bootstrap/client/internal/testutil"
+	"github.com/stretchr/testify/assert"
 )
 
-var _ = Describe("grpc", Ordered, func() {
+func TestGRPC(t *testing.T) {
 	var (
 		clusterCACertPEM []byte
-	)
 
-	BeforeAll(func() {
-		var err error
-		clusterCACertPEM, _, err = testutil.GenerateCertPEM(testutil.CertTemplate{
-			CommonName:   "hcp",
-			Organization: "aks",
-			IsCA:         true,
-			Expiration:   time.Now().Add(time.Hour),
-		})
-		Expect(err).To(BeNil())
+	)
+	var err error
+	clusterCACertPEM, _, err = testutil.GenerateCertPEM(testutil.CertTemplate{
+		CommonName:   "hcp",
+		Organization: "aks",
+		IsCA:         true,
+		Expiration:   time.Now().Add(time.Hour),
 	})
-
-	Context("secureTLSBootstrapServiceClientFactory", func() {
-		When("cluster ca data cannot be read", func() {
-			It("should return an error", func() {
-				serviceClient, close, err := getServiceClient("token", &Config{
-					ClusterCAFilePath: "does/not/exist.crt",
-					NextProto:         "nextProto",
-					APIServerFQDN:     "fqdn",
-				})
-
-				Expect(serviceClient).To(BeNil())
-				Expect(close).To(BeNil())
-				Expect(err).ToNot(BeNil())
-				Expect(err.Error()).To(ContainSubstring("reading cluster CA data from does/not/exist.crt"))
-
-			})
-		})
-		When("cluster ca data is invalid", func() {
-			It("should return an error", func() {
-				tempDir := GinkgoT().TempDir()
-				caFilePath := filepath.Join(tempDir, "ca.crt")
-				err := os.WriteFile(caFilePath, []byte("SGVsbG8gV29ybGQh"), os.ModePerm)
-				Expect(err).To(BeNil())
-
-				serviceClient, close, err := getServiceClient("token", &Config{
-					ClusterCAFilePath: caFilePath,
-					NextProto:         "nextProto",
-					APIServerFQDN:     "fqdn",
-				})
-
-				Expect(serviceClient).To(BeNil())
-				Expect(close).To(BeNil())
-				Expect(err).ToNot(BeNil())
-				Expect(err.Error()).To(ContainSubstring("failed to get TLS config"))
-				Expect(err.Error()).To(ContainSubstring("unable to construct new cert pool using cluster CA data"))
-			})
+	assert.NoError(t, err)
+	
+	t.Run("cluster ca data cannot be read", func(t *testing.T) {
```

Comment:
these cases currently invoked via t.Run at the top-level here can all be put into table testing format which is preferable, each table entry will just need to have its own preparation function which prepares the the mock CA data and listener (depending on the test case)